httprate.go

```diff
@@ -11,7 +11,12 @@ func Limit(requestLimit int, windowLength time.Duration, keyFuncs ...KeyFunc) fu
 	return NewRateLimiter(requestLimit, windowLength, nil, keyFuncs...).Handler
 }
 
+func LimitCustom(requestLimit int, windowLength time.Duration, onLimit OnLimitFunc, keyFuncs ...KeyFunc) func(next http.Handler) http.Handler {
```

Comment:
Let's find a different name than "LimitCustom". The "Custom" name doesn't feel right to me. I think perhaps instead this can go right on the "Limit()" method as a default argument option